pkg/apis/dmi/v1beta1/api.proto

```diff
@@ -0,0 +1,349 @@
+    // RemoveDevice unregisters a device to the device mapper.
+    // Device manager unregisters a device instance with the name of device
+    // to the mapper through the interface of RemoveDevice.
+    // When the mapper gets the request of unregister with device name,
+    // it should remove the device from the device list and disconnect to the real physical device.
+    rpc RemoveDevice(RemoveDeviceRequest) returns (RemoveDeviceResponse) {}
+    // UpdateDevice updates a device to the device mapper
+    // Device manager updates the information of a device used by the mapper
+    // through the interface of UpdateDevice.
+    // The information of a device includes the meta data and the status data of a device.
+    // When the mapper gets the request of updating with the information of a device,
+    // it should update the device of the device list and connect to the real physical device via the updated information.
+    rpc UpdateDevice(UpdateDeviceRequest) returns (UpdateDeviceResponse) {}
+    // CreateDeviceModel creates a device model to the device mapper.
+    // Device manager sends the information of device model to the mapper
+    // through the interface of CreateDeviceModel.
+    // When the mapper gets the request of creating with the information of device model,
+    // it should create a new device model to the list of device models.
+    rpc CreateDeviceModel(CreateDeviceModelRequest) returns (CreateDeviceModelResponse) {}
+    // RemoveDeviceModel remove a device model to the device mapper.
+    // Device manager sends the name of device model to the mapper
+    // through the interface of RemoveDeviceModel.
+    // When the mapper gets the request of removing with the name of device model,
+    // it should remove the device model to the list of device models.
+    rpc RemoveDeviceModel(RemoveDeviceModelRequest) returns (RemoveDeviceModelResponse) {}
+    // UpdateDeviceModel update a device model to the device mapper.
+    // Device manager sends the information of device model to the mapper
+    // through the interface of UpdateDeviceModel.
+    // When the mapper gets the request of updating with the information of device model,
+    // it should update the device model to the list of device models.
+    rpc UpdateDeviceModel(UpdateDeviceModelRequest) returns (UpdateDeviceModelResponse) {}
+    // UpdateDeviceStatus update a device status to the device mapper
+    // Device manager sends the new device status to the mapper
+    // through the interface of UpdateDeviceStatus.
+    // The device status represents the properties of device twins.
+    // When the mapper gets the request of updating with the new device status,
+    // it should update the device status of the device list and the real device status of the physical device via the updated information.
+    rpc UpdateDeviceStatus(UpdateDeviceStatusRequest) returns (UpdateDeviceStatusResponse) {}
```

Comment:
whethe the device status will include status other than properties values?

Reply:
yes, we will add something after.